frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -0,0 +1,113 @@
+import './_volume-snapshot-modal.scss';
+
+import * as React from 'react';
+
+import {
+  Dropdown,
+  HandlePromiseProps,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { Form, FormGroup, TextInput } from '@patternfly/react-core';
+import { K8sResourceKind, k8sCreate } from '@console/internal/module/k8s';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { VolumeSnapshotModel } from '../../../models';
+
+export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
+  const {
+    close,
+    cancel,
+    resourceName,
+    resourceNamespace,
+    errorMessage,
+    inProgress,
+    handlePromise,
+  } = props;
+  const [submitDisabled, setSubmitDisabled] = React.useState(false);
+  const [snapshotName, setSnapshotName] = React.useState(`${resourceName || 'pvc'}-snapshot-1`);
+  const snapshotTypes = {
+    single: 'Single Snapshot',
+  };
+
+  const onSnapshotNameChange = (value: string) => {
+    setSubmitDisabled(!value);
+    setSnapshotName(value);
+  };
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    const ns = resourceNamespace;
+    const pvcName = resourceName;
+    const snapshotTemplate: K8sResourceKind = {
+      apiVersion: VolumeSnapshotModel.apiVersion,
+      kind: VolumeSnapshotModel.kind,
+      metadata: {
+        name: snapshotName,
+        namespace: ns,
+      },
+      spec: {
+        source: {
+          name: pvcName,
+          kind: PersistentVolumeClaimModel.kind,
+        },
+      },
+    };
+    handlePromise(k8sCreate(VolumeSnapshotModel, snapshotTemplate))
+      .then(close)
+      .catch((error) => {
+        setSubmitDisabled(true);
+        throw error;
```

Comment:
not required

Reply:
We need to throw error to tell the user what went wrong.